sundials: 5.3.0 -> 5.6.1

doronbehar opened the pull request:
Update some cmake flags and be more explicit regarding index size.<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Update.

###### Things done

Tested `octave` to build with this update.

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

doronbehar committed:
sundials: 5.3.0 -> 5.6.1

Update some cmake flags and be more explicit regarding index size.

doronbehar committed:
sundials: Split outputs for examples

SuperSandro2000 commented on pkgs/development/libraries/sundials/default.nix:
```diff
@@ -30,37 +30,33 @@ stdenv.mkDerivation rec {
   };
 
-  patches = [
-    (fetchurl {
```
Please remove fetchurl from inputs.

doronbehar replied:
Needed to fetch the main src.

SuperSandro2000 commented:
Result of `nixpkgs-review pr 107286` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>python37Packages.scikits-odes</li>
    <li>python38Packages.scikits-odes</li>
    <li>python39Packages.scikits-odes</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>sundials</li>
  </ul>
</details>

```
    return _bootstrap._gcd_import(name[level:], package, level)
test_on_funcs_ida.py:16: in <module>
    from scikits.odes.sundials.ida import StatusEnumIDA
E   ImportError: dlopen(/nix/store/zckarm9nvz67fgg08w41b00nlf381ih3-python3.8-scikits.odes-2.6.1/lib/python3.8/site-packages/scikits/odes/sundials/ida.cpython-38-darwin.so, 2): Library not loaded: /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib/libsundials_ida.5.dylib
E     Referenced from: /nix/store/zckarm9nvz67fgg08w41b00nlf381ih3-python3.8-scikits.odes-2.6.1/lib/python3.8/site-packages/scikits/odes/sundials/ida.cpython-38-darwin.so
E     Reason: image not found
_______________ ERROR collecting test_user_return_vals_cvode.py ________________
ImportError while importing test module '/nix/store/zckarm9nvz67fgg08w41b00nlf381ih3-python3.8-scikits.odes-2.6.1/lib/python3.8/site-packages/scikits/odes/tests/test_user_return_vals_cvode.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/c56kq0k3f3rdgkj3s1z5hw0j4ack9gdg-python3-3.8.6/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test_user_return_vals_cvode.py:5: in <module>
    from ..sundials.cvode import StatusEnum
E   ImportError: dlopen(/nix/store/zckarm9nvz67fgg08w41b00nlf381ih3-python3.8-scikits.odes-2.6.1/lib/python3.8/site-packages/scikits/odes/sundials/cvode.cpython-38-darwin.so, 2): Library not loaded: /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib/libsundials_cvode.5.dylib
E     Referenced from: /nix/store/zckarm9nvz67fgg08w41b00nlf381ih3-python3.8-scikits.odes-2.6.1/lib/python3.8/site-packages/scikits/odes/sundials/cvode.cpython-38-darwin.so
E     Reason: image not found
________________ ERROR collecting test_user_return_vals_ida.py _________________
ImportError while importing test module '/nix/store/zckarm9nvz67fgg08w41b00nlf381ih3-python3.8-scikits.odes-2.6.1/lib/python3.8/site-packages/scikits/odes/tests/test_user_return_vals_ida.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/c56kq0k3f3rdgkj3s1z5hw0j4ack9gdg-python3-3.8.6/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test_user_return_vals_ida.py:5: in <module>
    from ..sundials.ida import StatusEnumIDA
E   ImportError: dlopen(/nix/store/zckarm9nvz67fgg08w41b00nlf381ih3-python3.8-scikits.odes-2.6.1/lib/python3.8/site-packages/scikits/odes/sundials/ida.cpython-38-darwin.so, 2): Library not loaded: /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib/libsundials_ida.5.dylib
E     Referenced from: /nix/store/zckarm9nvz67fgg08w41b00nlf381ih3-python3.8-scikits.odes-2.6.1/lib/python3.8/site-packages/scikits/odes/sundials/ida.cpython-38-darwin.so
E     Reason: image not found
=========================== short test summary info ============================
ERROR test_dae.py
ERROR test_odeint.py
ERROR test_on_funcs.py
ERROR test_on_funcs_ida.py
ERROR test_user_return_vals_cvode.py
ERROR test_user_return_vals_ida.p
```

markuskowa commented:
nixpkgs-review on x86_64-linux runs without errors.

doronbehar commented:
> Result of `nixpkgs-review pr 107286` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
> 3 packages failed to build:
> 1 package built:
> 
> ```
>     return _bootstrap._gcd_import(name[level:], package, level)
> test_on_funcs_ida.py:16: in <module>
>     from scikits.odes.sundials.ida import StatusEnumIDA
> E   ImportError: dlopen(/nix/store/zckarm9nvz67fgg08w41b00nlf381ih3-python3.8-scikits.odes-2.6.1/lib/python3.8/site-packages/scikits/odes/sundials/ida.cpython-38-darwin.so, 2): Library not loaded: /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib/libsundials_ida.5.dylib
> E     Referenced from: /nix/store/zckarm9nvz67fgg08w41b00nlf381ih3-python3.8-scikits.odes-2.6.1/lib/python3.8/site-packages/scikits/odes/sundials/ida.cpython-38-darwin.so
> E     Reason: image not found
> _______________ ERROR collecting test_user_return_vals_cvode.py ________________
> ImportError while importing test module '/nix/store/zckarm9nvz67fgg08w41b00nlf381ih3-python3.8-scikits.odes-2.6.1/lib/python3.8/site-packages/scikits/odes/tests/test_user_return_vals_cvode.py'.
> Hint: make sure your test modules/packages have valid Python names.
> Traceback:
> /nix/store/c56kq0k3f3rdgkj3s1z5hw0j4ack9gdg-python3-3.8.6/lib/python3.8/importlib/__init__.py:127: in import_module
>     return _bootstrap._gcd_import(name[level:], package, level)
> test_user_return_vals_cvode.py:5: in <module>
>     from ..sundials.cvode import StatusEnum
> E   ImportError: dlopen(/nix/store/zckarm9nvz67fgg08w41b00nlf381ih3-python3.8-scikits.odes-2.6.1/lib/python3.8/site-packages/scikits/odes/sundials/cvode.cpython-38-darwin.so, 2): Library not loaded: /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib/libsundials_cvode.5.dylib
> E     Referenced from: /nix/store/zckarm9nvz67fgg08w41b00nlf381ih3-python3.8-scikits.odes-2.6.1/lib/python3.8/site-packages/scikits/odes/sundials/cvode.cpython-38-darwin.so
> E     Reason: image not found
> ________________ ERROR collecting test_user_return_vals_ida.py _________________
> ImportError while importing test module '/nix/store/zckarm9nvz67fgg08w41b00nlf381ih3-python3.8-scikits.odes-2.6.1/lib/python3.8/site-packages/scikits/odes/tests/test_user_return_vals_ida.py'.
> Hint: make sure your test modules/packages have valid Python names.
> Traceback:
> /nix/store/c56kq0k3f3rdgkj3s1z5hw0j4ack9gdg-python3-3.8.6/lib/python3.8/importlib/__init__.py:127: in import_module
>     return _bootstrap._gcd_import(name[level:], package, level)
> test_user_return_vals_ida.py:5: in <module>
>     from ..sundials.ida import StatusEnumIDA
> E   ImportError: dlopen(/nix/store/zckarm9nvz67fgg08w41b00nlf381ih3-python3.8-scikits.odes-2.6.1/lib/python3.8/site-packages/scikits/odes/sundials/ida.cpython-38-darwin.so, 2): Library not loaded: /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib/libsundials_ida.5.dylib
> E     Referenced from: /nix/store/zckarm9nvz67fgg08w41b00nlf381ih3-python3.8-scikits.odes-2.6.1/lib/python3.8/site-packages/scikits/odes/sundials/ida.cpython-38-darwin.so
> E     Reason: image not found
> =========================== short test summary info ============================
> ERROR test_dae.py
> ERROR test_odeint.py
> ERROR test_on_funcs.py
> ERROR test_on_funcs_ida.py
> ERROR test_user_return_vals_cvode.py
> ERROR test_user_return_vals_ida.p
> ```

cc @flokli per https://github.com/bmcage/odes/issues/107#issuecomment-526141373 / https://github.com/LLNL/sundials/pull/19 ?

flokli commented:
Ouch, I don't have the hardware to properly test this on darwin. It might be this regressed upstream (not sure if there's any darwin testing on their side).

@rb2k, interested to take a look?

rb2k commented:
As is, there's some other problems when running nixpkgs-review pr 107286 from my side:
```
$ nix-env -f /Users/mseeger/.cache/nixpkgs-review/pr-107286/nixpkgs -qaP --xml --out-path --show-trace --meta
error: --- UndefinedVarError -------------------------------------------------------------------------------------------------------------------------------------------------------------------------- nix-env
at: (36:9) in file: /Users/mseeger/.cache/nixpkgs-review/pr-107286/nixpkgs/pkgs/development/libraries/sundials/default.nix

    35| 
    36|   src = fetchurl {
      |         ^
    37|     url = "https://computation.llnl.gov/projects/${pname}/download/${pname}-${version}.tar.gz";

undefined variable 'fetchurl'
------------------------------------------------------------------------------------------------- show-trace --------------------------------------------------------------------------------------------------
trace: while evaluating 'makeOverridable'
at: (67:24) in file: /Users/mseeger/.cache/nixpkgs-review/pr-107286/nixpkgs/lib/customisation.nix

    66|   */
    67|   makeOverridable = f: origArgs:
      |                        ^
    68|     let

trace: from call site
at: (121:8) in file: /Users/mseeger/.cache/nixpkgs-review/pr-107286/nixpkgs/lib/customisation.nix

   120|       auto = builtins.intersectAttrs (lib.functionArgs f) autoArgs;
   121|     in makeOverridable f (auto // args);
      |        ^
   122| 

trace: while evaluating 'callPackageWith'
at: (117:35) in file: /Users/mseeger/.cache/nixpkgs-review/pr-107286/nixpkgs/lib/customisation.nix

   116|   */
   117|   callPackageWith = autoArgs: fn: args:
      |                                   ^
   118|     let

trace: from call site
at: (16191:14) in file: /Users/mseeger/.cache/nixpkgs-review/pr-107286/nixpkgs/pkgs/top-level/all-packages.nix

 16190| 
 16191|   sundials = callPackage ../development/libraries/sundials {
      |              ^
 16192|     python = python3;

$ git worktree prune
Traceback (most recent call last):
  File "/nix/store/vn4v02vzwq8bslsy4ifch686napfsa0j-nixpkgs-review-2.5.0/bin/.nixpkgs-review-wrapped", line 9, in <module>
    sys.exit(main())
  File "/nix/store/vn4v02vzwq8bslsy4ifch686napfsa0j-nixpkgs-review-2.5.0/lib/python3.8/site-packages/nixpkgs_review/__init__.py", line 10, in main
    cli.main(command, args)
  File "/nix/store/vn4v02vzwq8bslsy4ifch686napfsa0j-nixpkgs-review-2.5.0/lib/python3.8/site-packages/nixpkgs_review/cli/__init__.py", line 228, in main
    args.func(args)
  File "/nix/store/vn4v02vzwq8bslsy4ifch686napfsa0j-nixpkgs-review-2.5.0/lib/python3.8/site-packages/nixpkgs_review/cli/pr.py", line 60, in pr_command
    contexts.append((pr, review.build_pr(pr)))
  File "/nix/store/vn4v02vzwq8bslsy4ifch686napfsa0j-nixpkgs-review-2.5.0/lib/python3.8/site-packages/nixpkgs_review/review.py", line 201, in build_pr
    return self.build_commit(base_rev, pr_rev)
  File "/nix/store/vn4v02vzwq8bslsy4ifch686napfsa0j-nixpkgs-review-2.5.0/lib/python3.8/site-packages/nixpkgs_review/review.py", line 155, in build_commit
    merged_packages = list_packages(str(self.worktree_dir()), check_meta=True)
  File "/nix/store/vn4v02vzwq8bslsy4ifch686napfsa0j-nixpkgs-review-2.5.0/lib/python3.8/site-packages/nixpkgs_review/review.py", line 303, in list_packages
    return parse_packages_xml(nix_env.stdout)
  File "/nix/store/vn4v02vzwq8bslsy4ifch686napfsa0j-nixpkgs-review-2.5.0/lib/python3.8/site-packages/nixpkgs_review/review.py", line 244, in parse_packages_xml
    for (event, elem) in context:
  File "/nix/store/0mabx9absir73mmijrsv0dsyvyya463r-python3-3.8.6/lib/python3.8/xml/etree/ElementTree.py", line 1233, in iterator
    root = pullparser._close_and_return_root()
  File "/nix/store/0mabx9absir73mmijrsv0dsyvyya463r-python3-3.8.6/lib/python3.8/xml/etree/ElementTree.py", line 1280, in _close_and_return_root
    root = self._parser.close()
xml.etree.ElementTree.ParseError: no element found: line 1, column 0
```

doronbehar commented:
> I think the last force-push might have been a bit dicey:

Correct. I removed the `fetchurl` from the arguments by mistake.

I removed the patch from our expression although that PR was closed because upstream wrote that it was included in v5.5.0. I didn't find the cmake variables mentioned in the PR in the main `CMakeLists.txt` but I did find them here:

https://github.com/LLNL/sundials/blob/5cf0b639797d12acd26ccd8459839ac5784abe04/cmake/SundialsSetupCompilers.cmake#L43-L62

Could it be we should set `BUILD_SHARED_LIBS` on darwin?

rb2k commented:
Result of `nixpkgs-review pr 107286` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>octave-jit</li>
  </ul>
</details>
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>octaveFull</li>
    <li>python37Packages.scikits-odes</li>
    <li>python38Packages.scikits-odes</li>
    <li>python39Packages.scikits-odes</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>gama</li>
    <li>minc_widgets</li>
    <li>octave</li>
    <li>sundials</li>
  </ul>
</details>

idontgetoutmuch commented:
I am not sure how to test this. I tried

```
nix-shell -I nixpkgs=/Users/dom/nixpkgs -p sundials
```

but got

```
strip is /nix/store/i1mccwrczhhacpk0jn58wmixqiqh1vkb-cctools-binutils-darwin-949.0.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib 
strip: can't process non-object and non-archive file: /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib/cmake/sundials/SUNDIALSTargets.cmake
strip: can't process non-object and non-archive file: /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib/cmake/sundials/SUNDIALSConfigVersion.cmake
strip: can't process non-object and non-archive file: /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib/cmake/sundials/SUNDIALSConfig.cmake
strip: can't process non-object and non-archive file: /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib/cmake/sundials/SUNDIALSTargets-release.cmake
patching script interpreter paths in /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1
strip is /nix/store/i1mccwrczhhacpk0jn58wmixqiqh1vkb-cctools-binutils-darwin-949.0.1/bin/strip
patching script interpreter paths in /nix/store/yxvhx5hm0jhcjpsmdj706kwywicaiwlf-sundials-5.6.1-examples
bash: direnv: command not found
```

I tried

```
cd /nix/store/yxvhx5hm0jhcjpsmdj706kwywicaiwlf-sundials-5.6.1-examples/share/examples/arkode/C_serial/
make
```

but got

```
[nix-shell:/nix/store/yxvhx5hm0jhcjpsmdj706kwywicaiwlf-sundials-5.6.1-examples/share/examples/arkode/C_serial]$ make
make
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_analytic.c
error: unable to open output file 'ark_analytic.o': 'Permission denied'
1 error generated.
make: *** [Makefile:90: ark_analytic.o] Error 1
```

doronbehar commented:
> Result of `nixpkgs-review pr 107286` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
> 1 package marked as broken and skipped:
> 4 packages failed to build:
> 
>     * octaveFull
> 
>     * python37Packages.scikits-odes
> 
>     * python38Packages.scikits-odes
> 
>     * python39Packages.scikits-odes
> 
> 
> 4 packages built:
> 
>     * gama
> 
>     * minc_widgets
> 
>     * octave
> 
>     * sundials

Thanks for the report @rb2k. Interesting `octaveFull` doesn't build but `octave` does. Maybe @graham33 could please verify that it's not due to this patch? I suspect something qt related on makes `octaveFull` fail to build on Darwin.

Also, I'd like to know if adding a cmake flag `BUILD_SHARED_LIBS` for sundials (on Darwin) helps scikit-odes for instance build with this update.

doronbehar commented:
> bash: direnv: command not found

That's weird and seems unrelated to me - you just spawn a `nix-shell` so perhaps your Bash rc files have a `direnv` initialization command in them while you don't use `direnv` anymore?

> I tried
> 
> ```
> cd /nix/store/yxvhx5hm0jhcjpsmdj706kwywicaiwlf-sundials-5.6.1-examples/share/examples/arkode/C_serial/
> make
> ```
> 
> but got
> 
> ```
> [nix-shell:/nix/store/yxvhx5hm0jhcjpsmdj706kwywicaiwlf-sundials-5.6.1-examples/share/examples/arkode/C_serial]$ make
> make
> /nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_analytic.c
> error: unable to open output file 'ark_analytic.o': 'Permission denied'
> ```

That makes sense - you can't run commands that create files in directories in `/nix/store`. Maybe try:

```
cp -r /nix/store/yxvhx5hm0jhcjpsmdj706kwywicaiwlf-sundials-5.6.1-examples/share/examples/arkode/C_serial sundials-C_serial-example
cd sundials-C_serial-example
make
```

idontgetoutmuch commented:
> > bash: direnv: command not found
> 
> That's weird and seems unrelated to me - you just spawn a `nix-shell` so perhaps your Bash rc files have a `direnv` initialization command in them while you don't use `direnv` anymore?

Yes probably something I missed when upgrading to Big Sur

> > I tried
> > ```
> > cd /nix/store/yxvhx5hm0jhcjpsmdj706kwywicaiwlf-sundials-5.6.1-examples/share/examples/arkode/C_serial/
> > make
> > ```
> > 
> > 
> > but got
> > ```
> > [nix-shell:/nix/store/yxvhx5hm0jhcjpsmdj706kwywicaiwlf-sundials-5.6.1-examples/share/examples/arkode/C_serial]$ make
> > make
> > /nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_analytic.c
> > error: unable to open output file 'ark_analytic.o': 'Permission denied'
> > ```
> 
> That makes sense - you can't run commands that create files in directories in `/nix/store`. Maybe try:
> 
> ```
> cp -r /nix/store/yxvhx5hm0jhcjpsmdj706kwywicaiwlf-sundials-5.6.1-examples/share/examples/arkode/C_serial sundials-C_serial-example
> cd sundials-C_serial-example
> make
> ```

Ok the examples build but I get segfaults - I am guessing the linker is picking up some incompatible versions of the libs - I continue to investigate.

idontgetoutmuch commented:
Ok I removed the brew versions and now I get

```
dom@MacBook-Pro sundials-C_serial-example % make
make
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_analytic.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_analytic_nonlin.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_brusselator.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_brusselator_fp.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_brusselator1D.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_heat1D.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_heat1D_adapt.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_KrylovDemo_prec.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_robertson.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_robertson_constraints.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_robertson_root.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_brusselator_mri.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_onewaycouple_mri.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_twowaycouple_mri.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_reaction_diffusion_mri.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_brusselator_1D_mri.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_kpr_mri.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_analytic ark_analytic.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_analytic_nonlin ark_analytic_nonlin.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_brusselator ark_brusselator.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_brusselator_fp ark_brusselator_fp.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_brusselator1D ark_brusselator1D.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_heat1D ark_heat1D.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_heat1D_adapt ark_heat1D_adapt.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_KrylovDemo_prec ark_KrylovDemo_prec.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_robertson ark_robertson.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_robertson_constraints ark_robertson_constraints.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_robertson_root ark_robertson_root.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_brusselator_mri ark_brusselator_mri.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_onewaycouple_mri ark_onewaycouple_mri.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_twowaycouple_mri ark_twowaycouple_mri.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_reaction_diffusion_mri ark_reaction_diffusion_mri.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_brusselator_1D_mri ark_brusselator_1D_mri.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_kpr_mri ark_kpr_mri.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [examples] Error 1
```

`Makefile` contains

```
prefix       = /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1
exec_prefix  = /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1
includedir   = /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include
libdir       = /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib

CPP      = /nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang
CPPFLAGS = -O3 -DNDEBUG
CC       = /nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang
CFLAGS   = -O3 -DNDEBUG
LDFLAGS  = 
LIBS     =  -lm

LINKFLAGS = -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
```

I changed this to

```
prefix       = /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1
exec_prefix  = /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1
includedir   = /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include
libdir       = /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib

CPP      = /nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang
CPPFLAGS = -O3 -DNDEBUG
CC       = /nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang
CFLAGS   = -O3 -DNDEBUG
LDFLAGS  = 
LIBS     =  -lm

LINKFLAGS = -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
```

and now everything builds but I get runtime errors such as

```
dom@MacBook-Pro sundials-C_serial-example % ./ark_analytic
./ark_analytic
dyld: Library not loaded: /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib/libsundials_arkode.4.dylib
  Referenced from: /Users/dom/sundials-C_serial-example/./ark_analytic
  Reason: image not found
zsh: abort      ./ark_analytic
```

flokli commented:
@idontgetoutmuch please don't build sundials manually via make, or via `nix-shell`. There can be anything leaking in from your global environment. This says very few about whether the actual attribute does build or not - a `nix-build -A sundials` in a checkout of this would provide much more insight.

I'm not sure about the state of sandboxing on darwin - you might want to remove any globally installed libraries first.

doronbehar commented:
I think I found out why upstream has introduced this regression:

This line:

https://github.com/LLNL/sundials/blob/5cf0b639797d12acd26ccd8459839ac5784abe04/cmake/SundialsSetupCompilers.cmake#L50

is different than this:

https://github.com/LLNL/sundials/pull/19/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR48

Should I open a PR upstream?

doronbehar commented:
Never mind, I'll open the PR upstream. Justified by:

https://github.com/jtojnar/cmake-snips#assuming-cmake_install_dir-is-relative-path

doronbehar commented:
@idontgetoutmuch if you could please retest now, I haven't opened the PR upstream but I updated the PR here to include the patch I'll send to them at:

https://github.com/LLNL/sundials/compare/master...doronbehar:rpath_install_dir?expand=1

doronbehar committed:
sundials: reorder attributes

Per
https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/attribute-ordering.md

idontgetoutmuch commented:
Tested - works like a charm :)

doronbehar commented:
Great, Updated the PR with an updated comment linking to the PR upstream.

doronbehar commented:
If someone can run a final nixpkgs-review on darwin it'd be great :pray:.

rb2k commented:
> If someone can run a final nixpkgs-review on darwin it'd be great 🙏.

running it :)

rb2k commented:
Result of `nixpkgs-review pr 107286` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>octave-jit</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>octaveFull</li>
  </ul>
</details>
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>gama</li>
    <li>minc_widgets</li>
    <li>octave</li>
    <li>python37Packages.scikits-odes</li>
    <li>python38Packages.scikits-odes</li>
    <li>python39Packages.scikits-odes</li>
    <li>sundials</li>
  </ul>
</details>

doronbehar commented:
> Result of `nixpkgs-review pr 107286` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
> 1 package marked as broken and skipped:
> 1 package failed to build:
> 7 packages built:

Thanks. The fact that octaveFull fails to build while octave isn't is not a regression of this PR:

https://hydra.nixos.org/job/nixpkgs/nixpkgs-20.09-darwin/octaveFull.x86_64-darwin

So I merge this.
